[Lu Luoyi] iP

luo-git opened the pull request.

Commit:
Add Gradle support

luo-git committed:
Add TextFormatter

luo-git committed:
Implement Level-1

luo-git committed:
Fix source code style

luo-git committed:
Implement Level-2

luo-git committed:
Reorganise into interfaces

luo-git committed:
Implement Level-3

luo-git committed:
Minor code formatting

luo-git committed:
Add ToDo, Deadline, Event classes

luo-git committed:
Implement Level-4

luo-git committed:
Clean up unicode logo

luo-git committed:
Implement A-TextUiTesting

Write test cases

luo-git committed:
Add Duke specific exceptions

luo-git committed:
Write testcases for Level-5

luo-git committed:
Implement Level-5

luo-git committed:
Improve exception test cases

luo-git committed:
Update test cases for Level-6

luo-git committed:
Implement Level-6

luo-git committed:
Add delete function

Add 3 classes.
StorageManager class manages reading and writing from the file system.
TaskEncoder class encodes of task into string.
TaskDecoder class decodes string into task.

luo-git committed:
Add time and list by time functionality.

luo-git committed:
Merge branch 'branch-Level-7'

luo-git committed:
Merge branch 'branch-Level-8'

# Resolve Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Event.java
#	src/main/java/ITask.java
#	src/main/java/Task.java

luo-git committed:
Add Command classes.

Including parent Command class, DeadlineCommand, DeleteCommand, DoneCommand, EventCommand, ListCommand, ToDoCommand.

luo-git committed:
Implement more OOP classes.

luo-git committed:
Refactor code into packages

luo-git committed:
Add Main driver class

luo-git committed:
Add Ui and TimeWrapper JUnit Tests

luo-git committed:
Fix bug with Ui and ListCommand

Mainly bugs with newlines.

luo-git committed:
Refactor classes into correct folder

luo-git committed:
Package the App as a JAR file

luo-git committed:
Add missing JavaDoc

luo-git committed:
Improve coding style

luo-git committed:
Add find command

A find command takes in a search string and displays matching tasks.

luo-git committed:
Fix Ui bug

luo-git committed:
Merge branch 'branch-A-JavaDoc'

luo-git committed:
Merge branch 'branch-A-CodingStandard'

luo-git committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/luoyi/duke/ui/Ui.java

luo-git committed:
Add gradle support

luo-git committed:
Fix unicode bug

luo-git committed:
Add checkstyle configurations

youaremysky99 commented on src/main/java/luoyi/duke/data/task/Event.java:
```diff
@@ -0,0 +1,51 @@
+        return String.format("E|%d|%s|%s", isDone ? 1 : 0, description, time);
+    }
+
+    public boolean isSameTime(String date) {
+        return time.equals(TimeWrapper.getTimeWrapper(date));
+    }
+
```
Should we have Javadoc here?

youaremysky99 commented on src/main/java/luoyi/duke/data/Duke.java:
```diff
@@ -0,0 +1,115 @@
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int getNumTask() {
+        return list.size();
+    }
```
Should we change it to getNumTasks?

youaremysky99 commented on src/main/java/luoyi/duke/parser/Parser.java:
```diff
@@ -0,0 +1,81 @@
+package luoyi.duke.parser;
+
+import luoyi.duke.commands.*;
```
I think stars should be avoided

youaremysky99 commented on src/main/java/luoyi/duke/storage/TaskDecoder.java:
```diff
@@ -0,0 +1,58 @@
+     * @param taskString List of task in string representation.
+     * @return List of task objects.
+     */
+    public static List<ITask> decodeList(List<String> taskString) {
+        List<ITask> taskList = new ArrayList<>();
+        taskString.forEach(x -> taskList.add(decodeTask(x)));
+        return taskList;
+    }
```
Should we change taskString to taskStrings as it represents a list?

youaremysky99 commented on src/main/java/luoyi/duke/storage/TaskDecoder.java:
```diff
@@ -0,0 +1,58 @@
+     */
+    private static ITask decodeTask(String taskString) {
+        String[] list = taskString.split("\\|");
+        ITask task;
```
As in coding standard guidelines, all objects should be initialized when declared.

youaremysky99 reviewed (commented):
I like your coding style, very clean!

luo-git committed:
Change Duke to mutable

Immutable Duke poses problem of multiple return since a String has to be returned for GUI.

luo-git committed:
Add GUI for duke

luo-git committed:
Merge branch 'add-gradle-support'

luo-git committed:
Merge branch 'A-CheckStyle'

luo-git committed:
Merge branch 'GUI-branch'

# Resolved Conflicts:
#	src/main/java/luoyi/duke/common/TextFormatter.java
#	src/main/java/luoyi/duke/data/task/Task.java

divakarmal commented on src/main/java/luoyi/duke/parser/Parser.java:
```diff
@@ -0,0 +1,83 @@
+                throw new DukeIllegalArgumentException(
+                        Message.ERR_WRONG_TODO_CMD.toString());
+            }
+            String description = commandStr.split(" ", 2)[1];
+            return ToDoCommand.getToDoCommand(description);
+        } else if (commandStr.matches("^deadline.*")) {
+            // Found deadline command
+            if (!commandStr.matches("^deadline .* /by .*")) {
+                throw new DukeIllegalArgumentException(
+                        Message.ERR_WRONG_DEADLINE_CMD.toString());
+            }
+            String[] params = commandStr.split(" ", 2)[1].split(" /by ");
+            return DeadlineCommand.getDeadlineCommand(params[0], params[1]);
+        } else if (commandStr.matches("^event.*")) {
+            // Handle event command
+            if (!commandStr.matches("^event .* /at .*")) {
+                throw new DukeIllegalArgumentException(
+                        Message.ERR_WRONG_EVENT_CMD.toString());
+            }
+            String[] params = commandStr.split(" ", 2)[1].split(" /at ");
+            return EventCommand.getEventCommand(params[0], params[1]);
+        } else if (commandStr.matches("^delete.*")) {
+            // Handle delete command
+            if (!commandStr.matches("^delete -?[0-9]+$")) {
+                throw new DukeIllegalArgumentException(
+                        Message.ERR_WRONG_DELETE_CMD.toString());
+            }
+            int index = Integer.parseInt(commandStr.split(" ")[1]);
+            return DeleteCommand.getDeleteCommand(index);
+        } else if (commandStr.matches("^find.*")) {
+            // Handle find command
+            if (!commandStr.matches("^find .*")) {
+                throw new DukeIllegalArgumentException(
+                        Message.ERR_WRONG_DELETE_CMD.toString());
+            }
+            String searchString = commandStr.split(" ", 2)[1];
+            return FindCommand.getFindCommand(searchString);
+        } else if (commandStr.equals("bye")) {
+            return ByeCommand.getByeCommand();
+        }
```
Very clear and concise and easy to understand, good job

divakarmal commented on src/main/java/luoyi/duke/data/task/Event.java:
```diff
@@ -0,0 +1,51 @@
+ * An Event class.
+ * An Event is a task that start at a specific time and ends at a specific time.
+ */
+public class Event extends Task {
```
Maybe class can be named EventTask instead of Event, makes it more clear

divakarmal commented on src/main/java/luoyi/duke/data/Duke.java:
```diff
@@ -0,0 +1,122 @@
+ */
+public class Duke implements IDuke {
+    /** List for storing Tasks */
+    private final TaskList list;
```
naming it list can be ambiguous, maybe it can be called taskList to make it clearer

luo-git committed:
Fix checkstyle error

luo-git committed:
Add new image resources

luo-git committed:
Add custom image and text background

luo-git committed:
Clean up GUI

luo-git committed:
Add assertions

luo-git committed:
Improve code quality

luo-git committed:
Merge pull request #2 from luo-git/A-Assertions

Add assertions

luo-git committed:
Merge branch 'master' into A-CodeQuality

luo-git committed:
Merge pull request #1 from luo-git/A-CodeQuality

Improve code quality

luo-git committed:
Adjust GUI for better visibility

luo-git committed:
Add continuous integration

luo-git committed:
Merge remote-tracking branch 'origin/master'

luo-git committed:
Fix continuous integration checkstyle error

luo-git committed:
Add in-App guidance to users

luo-git committed:
Add help message for HelpCommand

luo-git committed:
Fix CLI input

luo-git committed:
Merge pull request #3 from luo-git/C-Help

Add in-App guidance to users

luo-git committed:
Add and refactor new abstract class TimedTask

A TimedTask is a task that has a time parameter.

luo-git committed:
Add and refactor StoringCommand

A StoringCommand supports storage of tasks.

luo-git committed:
Add sorting functionality

luo-git committed:
Merge pull request #4 from luo-git/C-Sort

C-Sort

luo-git committed:
Improve code quality

luo-git committed:
Add TimeParserTest

luo-git committed:
Standardize TimeParser format

luo-git committed:
Add ui screenshot

luo-git committed:
Add a screenshot gif

luo-git committed:
Add user guide

luo-git committed:
Set theme jekyll-theme-midnight

luo-git committed:
Set theme jekyll-theme-midnight

luo-git committed:
Set theme jekyll-theme-modernist

luo-git committed:
Set theme jekyll-theme-hacker

luo-git committed:
Update readme files

luo-git committed:
Merge branch 'master' of https://github.com/luo-git/ip

luo-git committed:
Fix bug due to disable window resize

luo-git committed:
Add acknowledgement

luo-git committed:
Fix grammar in readme
